students/andrey_zkn/lesson05/mailroom3.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+import sys
+
+donors = {'Alexander Pushkin' : [200, 100, 340],
+          'Mikhail Lermontov' : [300, 100],
+          'Leo Tolstoy' : [150, 250, 100],
+          'Fyodor Dostoevsky' : [125],
+          'Anton Chekhov' : [100, 250],
+          'Nikolai Gogol' : [325, 150]
+          }
+
+def menu_options(): 
+    print( "\n".join (("Please select from the following options:",
+          "1 - Send a Thank You letter to a single donar.",
+          "2 - Create a Report.",
+          "3 - Send letters to all donors.",
+          "4 - Quit",
+          " >>> ")))
+    prompt = input('')
+    return prompt
+
+
+def thank_you_message(donor):
+    message = (f'\nDear {donor},'
+               f'\n\nThank you for your generous donation of ${sum(donors.get(donor)):,.2f}.' 
+               '\nWe value your contribution and support.' 
+               '\n\nSincerely,\n\nNew Horizon Charity Director\n')   
+    return message
+
+                                      
+def send_thank_you():
+    mail_to = input ("Enter the full name of a donor or 'list' for current donors ")
+    while mail_to.lower() == "list":
+        for name in donors:
+            print(name)
+        mail_to = input ("Please enter a  full name of a donor ")        
+    amount = float(input ("Enter the donation amount "))
```

Comment:
should have exception handling here in a case user enters non-number value

Reply:
Added this exception handling here. Thank you for the hint. The mailroom3.py file has been resubmitted. Please see my new pull request.